superset/assets/src/dashboard/util/layoutHelper.js

```diff
@@ -0,0 +1,42 @@
+import {
+  TABS_TYPE,
+  CHART_TYPE,
+} from './componentTypes';
+
+import {
+  DASHBOARD_ROOT_ID,
+} from './constants';
+
+export function getChartIdsFromLayout(layout) {
+  return Object.values(layout).reduce((chartIds, value) => {
+    if (value && value.meta && value.meta.chartId) {
+      chartIds.push(value.meta.chartId);
+    }
+    return chartIds;
+  }, []);
+}
+
+export function findInitialPageLoadCharts(layout) {
+  if (!layout) {
+    return [];
+  }
+
+  let chartIds = [];
+  function doTraverse(node) {
+    const type = node.type;
```

Comment:
object destructuring is preferred, I'll add a lint rule for this. `const { type } = node;`